archive/demos/shiny-drum-machine-audio.js

```diff
@@ -0,0 +1,307 @@
+/* eslint require-jsdoc: "off" */
+
+import {INSTRUMENTS, freeze, clone} from './shiny-drum-machine-data.js';
+
+const AudioContext = window.AudioContext ?? window.webkitAudioContext;
+
+const context = new AudioContext();
+
+const LOOP_LENGTH = 16;
+const BEATS_PER_FULL_NOTE = 4;
+const VOLUMES = freeze([0, 0.3, 1]);
```

Comment:
Why do we call freeze()? Are we worried that this can be changed unexpectedly?

Reply:
During the clean up, I called freeze on CONSTANTS to find unexpected changes, yes. Now, I am confident that constants are not touched, but see no benefit in removing the freeze calls.

Reply:
If we are confident, I believe reducing things that we need to maintain is better.

I am not sure if this was from the original code or not - is this added with the refactoring?

Reply:
I added all `freeze` and `seal` calls during refactoring. The hardcoded volume array is from the original code.

Reply:
Thanks for the explanation! Let's leave this as is, but keep in mind that keeping things simple is better for the maintenance and the readability.